modules/cas_cache/classifier.c

```diff
@@ -221,6 +221,42 @@ static int _cas_cls_string_ctr(struct cas_classifier *cls,
 	return 0;
 }
 
+/* IO direction condition constructor. @data is expected to contain string
+ * translated to IO direction.
+ */
+static int _cas_cls_direction_ctr(struct cas_classifier *cls,
+		struct cas_cls_condition *c, char *data)
+{
+	size_t len;
+	struct cas_cls_numeric *ctx;
+
+	if (!data) {
+		CAS_CLS_MSG(KERN_ERR, "Missing IO direction specifier\n");
+		return -EINVAL;
+	}
+
+	len = strnlen(data, MAX_STRING_SPECIFIER_LEN);
+
+	ctx = kmalloc(sizeof(*ctx), GFP_KERNEL);
+	if (!ctx)
+		return -ENOMEM;
+	ctx->operator = cas_cls_numeric_eq;
+
+	if (strncmp("read", data, len) == 0) {
+		ctx->v_u64 = REQ_OP_READ;
+	} else if (strncmp("write", data, len) == 0) {
+		ctx->v_u64 = REQ_OP_WRITE;
```

Comment:
`REQ_OP_READ`/`REQ_OP_WRITE` are technically correct in this context, however in other places in code we use `READ`/`WRITE`, which are more generic Linux kernel macros for data direction. I think we may want to keep it consistent, one way or another.